### PR TITLE
Fix asm.cpu config loading

### PR DESCRIPTION
**Pull request by ickromwerk**
https://github.com/radare/radare2/issues/13888

### CODE REVIEW

File: libr/core/cconfig.c [hunk 1]
```diff
@@ -505,10 +505,16 @@ static int cb_asmarch(void *user, void *data) {
 		if (newAsmCPU) {
 			if (*newAsmCPU) {
 				char *nac = strdup (newAsmCPU);
-				char *comma = strchr (nac, ',');
-				if (comma) {
-					*comma = 0;
-					r_config_set (core->config, "asm.cpu", nac);
+				if(asm_cpu) {
+					if(!strstr(nac, asm_cpu)) {
```
Comment (radare) [inline libr/core/cconfig.c]: add spaces before (

**Comment by radare** [comment]
this change makes 20 tests to fail, can you confirm my order theory?

**Comment by ickromwerk** [comment]
Actually I though about evaluation order before lurking down more. It was first though that when you change arch CPU reset to default one. And this is quite logical that you can specify CPU variant during arch evaluation.

> Many tests fails
That's why I call it quick and dirty :) I have noticed strange behaviour too at the end of the day. Is it possible to know which tests failed?

**Comment by radare** [comment]
another option and easier to solve would be to just hold the value of asm.cpu when set while parsing the script, and run it again after loading the project. can you update the PR in this way?

**Comment by ickromwerk** [comment]
No. I can't update it that way.

I noticed that loading logic is quite complex. First load vars, initializes assembler, then for unknown reason asm.arch bumps. And more and more.

Doing it the way you ask increases complexity and creates spaghetti code.

It should work this way: arch initializes -> loads asm plugin -> initialize asm plugin -> initialize cpu. if 'asm.cpu' already set during asm plugin cpu initialization -- it should be used.

I will investigate that further. I think that there is no reason to haste here ;)

**Comment by radare** [comment]
+1 thanks

**Comment by radare** [comment]
ping?

**Comment by radare** [comment]
ehoo!

**Comment by ickromwerk** [comment]
I have a tight schedule right now because I missed deadline for work project and people are waiting till I finish it. So its hard to invest more time to r2 project now.

If this PR messing things a bit it iz safe to delete it and continue discussion in 'issues' section.

**Pull request closed without merging**